scipy/sparse/tests/test_construct.py

```diff
@@ -459,6 +475,19 @@ def test_vstack_matrix_or_array(self):
         assert isinstance(construct.vstack([coo_matrix(A), coo_array(B)]), sparray)
         assert isinstance(construct.vstack([coo_matrix(A), coo_matrix(B)]), spmatrix)
 
+    def test_vstack_1d_with_2d(self):
```

Comment:
@perimosocordiae @dschult This test, backported from gh-21108, fails on the maintenance branch because 1D sparse array support is not as complete on the older branch. How would you like me to proceed?

If I delete that test, the full suite passes locally. Given this information, I'm not sure if you want me to do that or back the entire patch out, etc., so let me know what you prefer.

Reply:
The reproducer for the regression reported in gh-21064 is not fixed in this branch, which is what gh-21108 aimed to do. So it looks to me like backing out the backport is better than deleting the failing test.

Of course if it's fixable by another backport that is even better. However, I don't think it's the end of the world if the backport is dropped, since it was a regression from 1.12.0 to 1.13.0 so that can't be fully repaired anyway - plus it's a pretty niche thing to do, so as regressions go it's not such a painful one.

Reply:
The changes in #21108 to vstack/hstack are more general than fixing only the reported regression. There may be other cases where folks want to input, e.g.  1D coo arrays into vstack, and the fix from #21108 avoids those problems. So I would prefer removing this test to removing the whole patch. BUT: see below where we add to the test and fix some similar bugs that should be backported too.

In working on this issue, I added a couple of other tests, some of which fail. I put those along with fixes in #21400. That PR should probably be backported with #21108 to allow it to both be tested more thoroughly and to pass those tests.